external-crates/move/tools/move-package/tests/test_sources/diamond_problem_dep_nested_dep_conflict/Move.toml

```diff
@@ -0,0 +1,24 @@
+# Dependency graph and an address substitution introduced by an external resolver (C's dependency on
+# A and ADep is externally resolved). Even thoughm both C and B rely on the same version of A, in
+# the externally resolved graph ADep is reached via a dependency with address substitution whereas
+# in internally resolved graph ADep is reached via a dependency without address substitution.
```

Comment:
Okay, so to say it back to make sure I understand, this error case happens when two immediate dependency sub-graphs contain edges like:

- `Root -> C -> ... -> A -[addr_subst = { X = AX }]-> ADep`
- `Root -> D -> ... -> A -[addr_subst = { X = BX }]-> ADep`

(Same `from` and `to` on the edge, but different dependency information).

The case I was worried about being treated like an error was this one:

- `Root -> C -> ... -> A -[addr_subst = { X = AX }]-> ADep`
- `Root -> D -> ... -> B -[addr_subst = { X = BX }]-> ADep`

(different addr_subst, but also different `from` on the edge), because this is not an error.

Also, to make sure I understand -- this is not necessarily specific to external resolution, right?  I'm thinking you could get this kind of inconsistency between two internally resolved dependency sub-graphs as well (let's say that `A` was a local package, and `C` and `D` were written at different times against different versions of that local package, so at one time, `A`'s source code really did refer to `AX` and at another time, to `BX`).